sandboxnu/graduatenu
# SWR Custom Hooks for useStudent and usePlan

# Description

This PR installs `swr`, and adds SWR custom hooks for student and plan. It streamlines the process of getting the `student` and `plan` from the backend by placing it in an SWR hook. This also allows for mutation of these objects, and optimistic updates.

## Type of change


# How Has This Been Tested?

I have created 3 additional pages on the frontend:
1. To act as a [login](http://localhost:3000/login) page.
2. To act as a page that uses [student](http://localhost:3000/testusestudent).
3. To act as a page that uses [plan](http://localhost:3000/useplan).

To test:
- Run `yarn packages/api-client build` to ensure that that the api-client is looking at localhost:3001
- Ensure docker is running
- Run `yarn dev:db:up`
- Run `yarn dev:v2`
- - Delete all instances of "Aryan Shah" on the db using the "Clear DB" button.
- "Register" on the frontend, and create a new plan.
- Test out the various usestudent/useplan links above to see that you are able to access `student` and `plan` when logged in.
- Return back to the main page, logout and test the above links again.
- Ensure that you are redirected to /login as you are not logged in to make those api requests.

# Checklist:
